src/ray/core_worker/store_provider/memory_store/memory_store.cc

```diff
@@ -0,0 +1,206 @@
+#include <condition_variable>
+#include "ray/common/ray_config.h"
+#include "ray/core_worker/context.h"
+#include "ray/core_worker/core_worker.h"
+#include "ray/core_worker/object_interface.h"
+#include "ray/core_worker/store_provider/memory_store_provider.h"
+
+namespace ray {
+
+/// A class that represents a `Get` reuquest.
+class GetRequest {
```

Comment:
This is for both Get and Wait requests now, right? It should be renamed to ObjectRequest I think.

Reply:
I think the `MemoryStore` can only provide `Get` semantics for simplicity. And `MemoryStoreProvider` can implement both `Get` and `Wait` based on `MemoryStore::Get`. 
 Does this make sense to you?

Reply:
Ah ok, yeah that's fine.